File: src/main/java/com/pubnub/api/models/consumer/pubsub/PNSignalResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;

import com.pubnub.api.MessageType;
import com.pubnub.api.SpaceId;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString(callSuper = true)
public class PNSignalResult extends MessageResult {
    public PNSignalResult(BasePubSubResult basePubSubResult, JsonElement message, MessageType messageType, SpaceId spaceId) {
        super(basePubSubResult, message, messageType, spaceId);
    }
}

File: kotlin-publish/src/main/java/com/pubnub/publish/dependencies/models/PNSignalResult.kt
package com.pubnub.publish.dependencies.models

import com.google.gson.JsonElement
import com.pubnub.publish.dependencies.SubscribeCallback

/**
 * Wrapper around a signal received in [SubscribeCallback.signal].
 */
data class PNSignalResult internal constructor(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement
) : MessageResult, PubSubResult by basePubSubResult
